File: tests/java/test-app/src/main/java/com/godaddy/asherah/testapp/utils/SessionFactoryGenerator.java
package com.godaddy.asherah.testapp.utils;

import static com.godaddy.asherah.testapp.testhelpers.Constants.DEFAULT_PRODUCT_ID;
import static com.godaddy.asherah.testapp.testhelpers.Constants.DEFAULT_SYSTEM_ID;

import com.godaddy.asherah.appencryption.SessionFactory;
import com.godaddy.asherah.testapp.TestSetup;

public final class SessionFactoryGenerator {

  private SessionFactoryGenerator() {

  }

  public static SessionFactory createDefaultSessionFactory() {
    return createDefaultSessionFactory(DEFAULT_PRODUCT_ID, DEFAULT_SYSTEM_ID);
  }

  public static SessionFactory createDefaultSessionFactory(final String productId,
                                                           final String systemId) {
    // Read volatile first to force other members to be read from memory
    if (!TestSetup.isInitialized()) {
      throw new IllegalStateException("initialization has not been run yet!");
    }

    return SessionFactory.newBuilder(productId, systemId)
        .withMetastorePersistence(TestSetup.getMetastorePersistence())
        .withNeverExpiredCryptoPolicy()
        .withKeyManagementService(TestSetup.getKeyManagementService())
        .withMetricsEnabled()
        .build();
  }
}

File: languages/csharp/AppEncryption/AppEncryption.IntegrationTests/Utils/SessionFactoryGenerator.cs
using GoDaddy.Asherah.AppEncryption.KeyManagement;
using GoDaddy.Asherah.AppEncryption.Persistence;
using Newtonsoft.Json.Linq;
using static GoDaddy.Asherah.AppEncryption.IntegrationTests.TestHelpers.Constants;

namespace GoDaddy.Asherah.AppEncryption.IntegrationTests.Utils
{
    public static class SessionFactoryGenerator
    {
        public static SessionFactory CreateDefaultSessionFactory(
            KeyManagementService keyManagementService, IMetastore<JObject> metastore)
        {
            return CreateDefaultSessionFactory(DefaultProductId, DefaultSystemId, keyManagementService, metastore);
        }

        private static SessionFactory CreateDefaultSessionFactory(
            string productId,
            string systemId,
            KeyManagementService keyManagementService,
            IMetastore<JObject> metastore)
        {
            return SessionFactory.NewBuilder(productId, systemId)
                .WithMetastore(metastore)
                .WithNeverExpiredCryptoPolicy()
                .WithKeyManagementService(keyManagementService)
                .Build();
        }
    }
}
